offset_request.go

```diff
@@ -12,7 +12,7 @@ func (r *offsetRequestBlock) encode(pe packetEncoder) error {
 }
 
 type OffsetRequest struct {
-	blocks map[string]map[int32]*offsetRequestBlock
+	blocks map[string]map[int32][]*offsetRequestBlock
```

Comment:
since there's no need for uniqueness per partition anymore, this can just become `map[string][]*offsetRequestBlock` and you can put a field `partition int32` into `offsetRequestBlock`.